pkg/cmd/kafka/list/list.go

```diff
@@ -185,6 +187,43 @@ func mapResponseItemsToRows(kafkas []kafkamgmtclient.KafkaRequest, selectedId st
 	return rows
 }
 
+func getClusterIdMapFromKafkas(opts *options, kafkas []kafkamgmtclient.KafkaRequest) (map[string]*v1.Cluster, error) {
+	kafkaClusterIds := make([]string, len(kafkas))
+	for _, kafka := range kafkas {
+		if kafka.GetClusterId() != "" {
+			kafkaClusterIds = append(kafkaClusterIds, kafka.GetClusterId())
+		}
+	}
+
+	idToCluster := make(map[string]*v1.Cluster)
+
+	// if no kafkas have a cluster id assigned then we can skip the call to get
+	// the clusters as we dont need their info
+	if len(kafkaClusterIds) > 0 {
+		clusterList, err := clustermgmt.GetClusterListByIds(opts.f, opts.clusterManagementApiUrl, opts.accessToken, createSearchString(kafkaClusterIds), len(idToCluster))
```

Comment:
```suggestion
		clusterList, err := clustermgmt.GetClusterListByIds(opts.f, opts.clusterManagementApiUrl, opts.accessToken, createSearchString(kafkaClusterIds), len(kafkaClusterIds))
```

`len(idToCluster)` is `0` at this stage as it is set afterward post this call.